gaussian_example/transform_base-1.0.yaml

```diff
@@ -0,0 +1,135 @@
+%YAML 1.1
+---
+$schema: http://stsci.edu/schemas/yaml-schema/draft-01
+id: asdf://asdf-format.org/extensions/transform/schemas/transform_base-1.0
+description: >
+  Common validations for transforms.
+
+definitions:
+  parameter_value:
+    description: >
+      Parameter value.  May be a single number, or in the case of a
+      transform set, a 1D ndarray of numbers.
+    anyOf:
+      - type: number
+      - tag: asdf://asdf-format.org/extensions/core/tags/ndarray-*
+        ndim: 1
+
+  parameter:
+    description: >
+      Object that encapsulates properties of a transform parameter.
+    properties:
+      value:
+        description: >
+          Parameter value.
+        $ref: '#/definitions/parameter_value'
+      unit:
+        description: >
+          Parameter unit.
+        default: >
+          Parameter has no unit.
+        tag: asdf://asdf-format.org/extensions/core/tags/unit-*
+      bounds:
+        description: >
+          Inclusive upper and lower bounds of the parameter value.
+          Used during fitting.
+        default: >
+          Parameter value is unbounded.
+        $ref: '#/definitions/bounds'
+      fixed:
+        description: >
+          If true, this parameter's value should be fixed when fitting.
+        default: >
+          Parameter is not fixed.
+        type: boolean
```

Comment:
It looks like we need a `parameter` definition if we want to save the constraints under each parameter instead of a global model definition of all constraints.
OTH we have also Model constraints, not just the Parameter constraints. Those will have to be defined for the model, i.e. we are splitting constraints definition in two places.
In addition, for polynomials, most likely, we will have constraints serialized not per parameter but as an array. But this is true also for their parameters. 

All fitting routines I'm familiar with have a separate argument or constraints. So from implementation point of view (and perhaps performance) it may be better to have the constraints separate from parameters.

Reply:
Is it feasible to represent all types of constraints the same way?  For fixed parameters, for example, set to a `constant-1.0` transform.